### PR TITLE
[AutoPlugin] support dg2

### PR DESCRIPTION
**Pull request by tiger100256-hu**
### Details:
 - *rebase the code to latest master,origin pr #7337*
 - *add unit test case*

### Tickets:
 - *63465*

### CODE REVIEW

**Review by wangleis — APPROVED** [review]

File: src/plugins/auto/plugin.cpp [hunk 1]
```diff
@@ -433,16 +434,19 @@ DeviceInformation MultiDeviceInferencePlugin::SelectDevice(const std::vector<Dev
             auto& gpuUniqueName = item.uniqueName;
             if (gpuUniqueName.find("iGPU") != std::string::npos) {
                 iGPU.push_back(item);
-            } else if (gpuUniqueName.find("dGPU") != std::string::npos) {
-                dGPU.push_back(item);
+            } else if (gpuUniqueName.find("0x4f87") != std::string::npos) {
+                dGPU_DG2.push_back(item);
+            } else if (gpuUniqueName.find("Intel(R) Iris(R) Xe MAX Graphics") != std::string::npos) {
```
Comment (vladimir-paramuzov) [inline src/plugins/auto/plugin.cpp:439]: please don't use device id or device names as it's not future proof an usually not portable. GPU_UARCH_VERSION and DEVICE_TYPE metrics can be used instead

**Review by vladimir-paramuzov — CHANGES_REQUESTED** [review]

File: src/plugins/auto/plugin.cpp (hunk 1 above)
Comment (myshevts) [reply src/plugins/auto/plugin.cpp:439]: thanks, @vladimir-paramuzov. Yes, here is how to recognize the DG2 https://github.com/openvinotoolkit/openvino/blob/9e41208791aecbf12a000f5b739683631e246551/src/plugins/intel_gpu/src/plugin/plugin.cpp#L661

**Comment by myshevts** [comment]
I presume this PR is not valid anymore? @tiger100256-hu  pls close it, if it is not relevant

**Comment by tiger100256-hu** [comment]
@myshevts ok, I will reopen it if need

**Pull request closed without merging**